app/assets/javascripts/discourse/app/components/date-picker.js

```diff
@@ -50,6 +54,14 @@ export default class DatePicker extends Component {
         };
 
         this._picker = new Pikaday(Object.assign(options, this._opts()));
+
+        if (this.value) {
+          const parsedDate =
+            this.value instanceof moment ? this.date : moment(this.value);
```

Comment:
```suggestion
            this.value instanceof moment ? this.value : moment(this.value);
```

Should be this no? Also I don't think this check is necessary, if you do `moment(value)` when `value` is already a moment object, it just returns it.

If you really need to check, you can do `moment.isMoment(this.value)`

Reply:
yes I think you are right, will check :+1: